crates/biome_js_analyze/src/lint/nursery/no_unwanted_polyfillio.rs

```diff
@@ -0,0 +1,273 @@
+    ///   <Script src='https://polyfill.io/v3/polyfill.min.js?features=IntersectionObserver' />
+    ///   <Script src='https://polyfill-fastly.io/v3/polyfill.min.js?features=IntersectionObserver' />
+    /// </>
+    /// ```
+    ///
+    pub NoUnwantedPolyfillio {
+        version: "next",
+        name: "noUnwantedPolyfillio",
+        language: "jsx",
+        sources: &[RuleSource::EslintNext("no-unwanted-polyfillio")],
+        source_kind: RuleSourceKind::SameLogic,
+        recommended: false,
+    }
+}
+
+pub struct RuleState {
+    range: TextRange,
+    unwanted_features_message: Box<str>,
+}
+
+impl Rule for NoUnwantedPolyfillio {
+    type Query = Semantic<AnyJsxElement>;
+    type State = RuleState;
+    type Signals = Option<Self::State>;
+    type Options = ();
+
+    fn run(ctx: &RuleContext<Self>) -> Self::Signals {
+        let jsx_element = ctx.query();
+        let element_name = jsx_element.name().ok()?;
+        let element_name = element_name.name_value_token().ok()?;
+
+        let src_attribute = jsx_element.attributes().find_by_name("src")?;
+        let src_attribute_value = src_attribute.initializer()?.value().ok()?;
+        let src_attribute_str = src_attribute_value
+            .as_jsx_string()?
+            .inner_string_text()
+            .ok()?;
+
+        let features = match element_name.text_trimmed() {
+            "script" => check_unwanted_polyfill(&src_attribute_str, &NEXT_POLYFILLED_FEATURES)?,
+            _ => {
+                let semantic_model = ctx.model();
+                let reference = jsx_element.name().ok()?;
+                let reference = reference.as_jsx_reference_identifier()?;
+                let binding = semantic_model.binding(reference)?;
+                if is_next_import(&binding, NextLibrary::Script) {
+                    check_unwanted_polyfill(&src_attribute_str, &NEXT_POLYFILLED_FEATURES)?
+                } else {
+                    return None;
+                }
+            }
+        };
+        Some(RuleState {
+            range: src_attribute_value.range(),
+            unwanted_features_message: unwanted_features_message(&features),
+        })
+    }
+
+    fn diagnostic(_ctx: &RuleContext<Self>, state: &Self::State) -> Option<RuleDiagnostic> {
+        Some(
+            RuleDiagnostic::new(
+                rule_category!(),
+                state.range,
+                markup! { "Prevent duplicate polyfills from Polyfill.io" },
+            )
+            .note(markup! { {state.unwanted_features_message} " already shipped with Next.js." })
+            .note(markup! { "This unnecessarily increases page weight which can affect loading performance." }),
+        )
+    }
+}
+
+// Keep in sync with next.js polyfills file : https://github.com/vercel/next.js/blob/master/packages/next-polyfill-nomodule/src/index.js
+const NEXT_POLYFILLED_FEATURES: [&str; 64] = [
```

Comment:
This way you don't need to keep the length up-to-date manually:

```suggestion
const NEXT_POLYFILLED_FEATURES: &[&str] = &[
```